Fix `rz_file_basename` on Windows

GustavoLCR opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

It was broken for paths with mixed `\` and `/` on windows

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

Unit test pass

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

None

GustavoLCR committed:
Fix `rz_file_basename` on Windows

XVilka commented:
```
-- stderr
ERROR: Cannot open file '/home/runner/.local/share/rizin/pdb/C:\Users\Segev\junk\exceptions-test\build\Debug\exceptions-test.pdb/2A7ED8A5D796443D80CF7200044EDF833/C:\Users\Segev\junk\exceptions-test\build\Debug\exceptions-test.pdb'
ERROR: Cannot find "_TP_CALLBACK_ENVIRON_V3" struct type



[XX] db/formats/pdb LF_INDEX parsing
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'idp
te CV_HREG_e~?
' bins/pdb/exceptions-test.exe
-- stdout
--- expected
+++ actual
@@ -1,1 +1,1 @@
-3218
+0

-- stderr
ERROR: Cannot open file '/home/runner/.local/share/rizin/pdb/C:\Users\Segev\junk\exceptions-test\build\Debug\exceptions-test.pdb/2A7ED8A5D796443D80CF7200044EDF833/C:\Users\Segev\junk\exceptions-test\build\Debug\exceptions-test.pdb'
ERROR: Cannot find "CV_HREG_e" enum type
```

wargio reviewed (approved):
this is the same issue i had with sigdb on windows.

GustavoLCR committed:
Introduce `rz_file_dos_basename()` API

Use it whenever we need to interact with paths that can contain `\` as a separator as well
